File: 01.C/problem_set/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_positive_change(void);
int return_number_of_coins(int change);

int main(void)
{
    int change = get_positive_change();
    int coin = return_number_of_coins(change);
    printf("%i\n", coin);
}

int get_positive_change(void)
{
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);
    return change;
}

int return_number_of_coins(int change)
{
    int no_coin = 0;

    // Quarters (25 cents)
    no_coin += change / 25;
    change %= 25;

    // Dimes (10 cents)
    no_coin += change / 10;
    change %= 10;

    // Nickels (5 cents)
    no_coin += change / 5;
    change %= 5;

    // Pennies (1 cent) - whatever remains
    no_coin += change;

    return no_coin;
}

File: 06.Python/problem_set/sentimental-cash/cash.py
from cs50 import get_float

change = get_float("Change: ")
num_coins = 0
if change < 0:
    change = get_float("Change: ")

if change >= 0.25:
    quarters = int(change // 0.25)
    num_coins += quarters
    change = round(change % 0.25, 2)
if change >= 0.10:
    dimes = int(change // 0.10)
    num_coins += dimes
    change = round(change % 0.10, 2)
if change >= 0.05:
    nickels = int(change // 0.05)
    num_coins += nickels
    change = round(change % 0.05, 2)
if change >= 0.01:
    pennies = change // 0.01
    num_coins += pennies

print(num_coins)
